contracts/allocators/alchemistAllocator.sol

```diff
@@ -0,0 +1,210 @@
+     *  @param isCompounding used to indicate if the contract is compounding pending rewards
+     */
+    function deposit(
+        uint256 amount,
+        uint256 _poolId,
+        bool isCompounding
+    ) public onlyOwner {
+        if (isCompounding) {} else {
+            treasury.manage(alchemist, amount); // retrieve amount of asset from treasury
+        }
+
+        IERC20(alchemist).approve(address(tALCX), amount); // approve tALCX pool to spend tokens
+        tALCX.deposit(amount);
+
+        totalAlchemistDeposited = totalAlchemistDeposited.add(amount);
+
+        uint256 tALCX_balance = IERC20(address(tALCX)).balanceOf(address(this));
+        require(tALCX_balance == amount, "received tALCX must be 1:1 ratio with deposited alchemist amount");
+
+        IERC20(address(tALCX)).approve(address(pool), tALCX_balance); // approve to deposit to Alchemist staking pool
+        pool.deposit(_poolId, tALCX_balance); // deposit into Alchemist staking pool
+    }
+
+    /**
+     *  @notice as structured by Tokemak before one withdraws you must first request withdrawal,
+            unstake tALCX from Alchemist staking pool and make a request on Tokemak tALCX pool.
+     *  @param _poolId pool id of tALCX on Alchemist staking pool
+     *  @param _amount amount to withdraw if _isEntireFunds is false
+     *  @param _isEntireFunds used to indicate if amount to with is the entire funds deposited
+     */
+    function requestWithdraw(
+        uint256 _poolId,
+        uint256 _amount,
+        bool _isEntireFunds
+    ) external onlyOwner {
+        if (_isEntireFunds) {
+            pool.exit(_poolId);
+        } else {
+            pool.withdraw(_poolId, _amount);
+        }
+
+        uint256 balance = IERC20(address(tALCX)).balanceOf(address(this));
+
+        tALCX.requestWithdrawal(balance);
+    }
+
+    /**
+     *  @notice withdraws ALCX from Tokemak tALCX pool then deposits asset into treasury,
+            ensures cycle for withdraw has been reached.
+     */
+    function withdraw() external onlyOwner {
+        (uint256 minCycle, ) = tALCX.requestedWithdrawals(address(this));
+        uint256 current_cycle = ITokemak_Manager(tokemak_manager).currentCycleIndex();
+
+        require(minCycle <= current_cycle, "requested withdraw cycle not reached yet");
+
+        (, uint256 requested_amount_to_withdraw) = getRequestedWithdrawalInfo();
+        tALCX.withdraw(requested_amount_to_withdraw);
+        uint256 balance = IERC20(alchemist).balanceOf(address(this)); // balance of asset withdrawn
+
+        // account for withdrawal
+        uint256 value = treasury.tokenValue(alchemist, balance);
+        totalAlchemistDeposited = totalAlchemistDeposited.sub(requested_amount_to_withdraw);
+
+        IERC20(alchemist).approve(address(treasury), balance); // approve to deposit asset into treasury
+        treasury.deposit(balance, alchemist, value); // deposit using value as profit so no OHM is minted
```

Comment:
Will just transfer into treasury, current treasury does not accept non reserve or liquidity tokens to depoist

Reply:
In the test case, I was able to enable respective addresses to be reserved tokens on the treasury.

Reply:
@dev-Cnote ALCX is not a reserve token for the treasury, so will not be able to utilize `deposit`